Automattic/jetpack
# Add/jetpack waf plugin hook

#### Changes proposed in this Pull Request:

* Runs the Jetpack WAF as part of the Jetpack plugin if you have set the `wp_option` set accordingly (see testing section below for details).

#### Jetpack product discussion



1201069996155224-as-1201751378613114

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






I have added a mock firewall rule that checks if `jetpack-waf-always-block` is part of the request's query string, and then blocks it accordingly.

(You can do all those tests both on the site itself and in wp-admin.)
* View the site with `?jetpack-waf-always-block` appended to the URL (or add it somewhere else in the query string).
* Verify that the request is not blocked (because we did not enable the feature yet)
* Add an option entry to enable the feature. E.g. by CLI: `wp option add jetpack_waf_mode normal`
* View the site with `?jetpack-waf-always-block` appended to the URL (or add it somewhere else in the query string).
* Verify that your request is now blocked
* Verify that your request is not blocked (works as normal) if you don't have that in your query string

#### Notes

I appreciate that all the added `// phpcs:ignore` are not really cool to do. This was however the only way I found to do it for now, since we want to be doing some unusual things here. Very open to input!

The "do nothing" error handling is probably also not what we want in the long run; we'll have to figure out what we are going to do with this.
